core/src/main/java/org/elasticsearch/cluster/routing/AllocationId.java

```diff
@@ -198,6 +201,6 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
     }
 
     public static AllocationId fromXContent(XContentParser parser) throws IOException {
-        return ALLOCATION_ID_PARSER.parse(parser, new AllocationId.Builder()).build();
+        return ALLOCATION_ID_PARSER.parse(parser, new AllocationId.Builder(), () -> ParseFieldMatcher.STRICT).build();
```

Comment:
I see how using strict is bw compatible as we had hardcoded STRICT in AbstractXContentParser. That said, that was wrong in the first place. We should make sure that we have some kind of parse context that is at least a ParseFieldMatcherSupplier every single time we parse anything. Callers of these static parse methods should provide it, and ideally these STRICT and EMPTY constants should be moved to test classes. Hopefully those callers will have settings available to understand whether we are in strict mode or not, or use the same defaults everywhere if settings are not available (not like we do now that we randomly use STRICT or EMPTY here and there). This can be a follow-up, it is a bigger change, but it seems easier and cleaner to achieve after this PR, which is great!